src/indieweb_utils/parsing/parse.py

```diff
@@ -0,0 +1,49 @@
+import mf2py
+import requests
+from bs4 import BeautifulSoup
+
+
+class RequestError(Exception):
+    pass
+
+
+def get_parsed_mf2_data(parsed_mf2: mf2py.Parser, html: str = "", url: str = ""):
+    """
+    Return or create an mf2py object from a parsed document, a HTML string, and a URL.
+    """
+    if parsed_mf2:
+        return parsed_mf2
+    elif html:
+        return mf2py.parse(doc=html)
+    elif url:
+        return mf2py.parse(url=url)
+
+    raise RequestError("No soup, url, or HTML document provided.")
+
+
+def _get_soup(html: str = "", soup: BeautifulSoup = BeautifulSoup, url: str = "", headers: dict = {}) -> BeautifulSoup:
```

Comment:
Prefixing a function with `_` means it should be private to that module i.e. not part of the "public" api. As we're using `_get_soup` in `get_profile`, we should rename this to `get_soup`.